src/CachePeer.h

```diff
@@ -32,6 +33,13 @@ class CachePeer
     CachePeer() = default;
     ~CachePeer();
 
+    /// reacts to a successful establishment of a TCP connection to this cache_peer
```

Comment:
I do not think the suggested wording reflects how the official and PR code actually work. The exact condition is difficult to define precisely, but these successes and failures are not about cache_peer _use_. They are about _establishment_ of a connection (or _preparation_ for use). As we polish how cache_peer failures are reported/detailed (another in-progress project), we should drop the TCP part. For now, let's keep it as an imperfect marker of a code that needs further work that is out of scope here.

Said that, if you insist, I will drop TCP from the description in hope to avoid blocking this PR.

Reply:
As I see it success should be marked if and only when either a) a probe completes whatever transaction is tried, or b) an HTTP transaction completed without 4xx/5xx status. Despite the great variance of technical fine-details hidden behind those statements - the term "use" is vague enough to cover them all without ambiguity. Less wide terms I can think of all seem to conflict or imply wrong things about some of the fine-detail. Better suggestions welcome.

I do insist on at least removing the tight association with TCP connection setup. That is a common case, but not the only one.

Reply:
> As I see it success should be marked if and only when either a) a probe completes whatever transaction is tried, or b) an HTTP transaction completed without 4xx/5xx status.

That may be the right _future_ principle, but it is not how the official code works (and is not how the official code was _designed_ to work -- so this is not just about bugs). In other words, the official code is using a different principle (which is a lot closer to TCP than to HTTP). This PR does not change any of that and should not change any of that.

> I do insist on at least removing the tight association with TCP connection setup.

Removed in 7aa16d6.